Remove legacy 0% = not targetable assumption

reaperrr opened the pull request:
None of the first-party mods use the above assumption anymore as far as I could tell, so we might as well remove it now.

reaperrr committed:
Remove legacy 0% = not targetable assumption

Mailaender commented:
Can you make the utility spit out a warning on upgrade?

GraionDilach commented:
You should remove the 0% verses if they are unused anyway.
https://github.com/OpenRA/OpenRA/blob/bleed/mods/ra/weapons/smallcaliber.yaml#L10
https://github.com/OpenRA/OpenRA/blob/bleed/mods/ra/weapons/smallcaliber.yaml#L290
https://github.com/OpenRA/OpenRA/blob/bleed/mods/ra/weapons/smallcaliber.yaml#L341
https://github.com/OpenRA/OpenRA/blob/bleed/mods/ra/weapons/other.yaml#L189

Also MadTank seemingly use it
https://github.com/OpenRA/OpenRA/blob/bleed/mods/ra/weapons/other.yaml#L281

reaperrr committed:
Fix D2k target validation

0% versus no longer means not targetable.

reaperrr commented:
> You should remove the 0% verses if they are unused anyway.

My bad, I messed up when searching for these cases (I forgot we use integer math now and no longer use the '%').

I'll update this later.

reaperrr committed:
TD target validation

This mod already implements ValidTargets as intended, but grounded helicopters should count as vehicles for target validation.

reaperrr committed:
Fix TS medic heal target validation

reaperrr committed:
Fix RA target validation

0% versus no longer means cannot target

reaperrr committed:
Upgrade notice

reaperrr commented:
Updated.

GraionDilach commented on mods/d2k/weapons.yaml:
```diff
@@ -66,19 +67,14 @@ Sniper:
 	Range: 5c768
 	Report: VULCAN.AUD
-	ValidTargets: Ground
+	InvalidTargets: Structure
```
Wouldn't this render the weapon anti-air capable?

reaperrr replied:
`ValidTargets` defaults to `Ground` and `Water`, you'll find that many normal weapons don't explicitly define `ValidTargets` at all, so that entry was simply redundant (though you could argue it removed `Water` since that doesn't exist in d2k).

penev92 replied:
Yay for random default values, eh?

reaperrr replied:
*shrugs* Legacy from when OpenRA was just RA, I guess. Might be worth dealing with at some point, but not in this PR.

abcdefg30 commented:
:+1:

abcdefg30 commented:
Oh wait, some units like `Tanya`, the `Sniper` or the `Spy` can't force fire on the ground anymore.

reaperrr commented:
Updated.

Fortunately that was just a yaml issue, the `Ground` target type must be valid to allow force-firing on ground.

GraionDilach commented on mods/d2k/weapons.yaml:
```diff
@@ -88,12 +85,10 @@ Vulcan:
 		Spread: 96
 		Damage: 30
-		ValidTargets: Ground
+		InvalidTargets: Buildings
```
It's Buildings or Structure?

penev92 replied:
Looks like it's "Structure" everywhere else.

reaperrr replied:
Good catch, fixed.

GraionDilach commented:
:+1: after my above nit gets fixed.

reaperrr commented:
Updated, can be merged now.

penev92 commented:
Changelog, anyone?
I feel this is somewhat important for modders to know about...

reaperrr commented:
[changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-%28bleed%29/_compare/82edd49b510786fedfbc7c7c4ccc01bbd41caa4e...f80567f63accba61f29b991376abe393a5b2c3e7)
